### DISCUSSION TITLE
PiVPN and other applications

### DISCUSSION THREAD

**Discussion by viertelb** [pivpn/pivpn#1586] [Q&A] [answered]
What does the script change in terms of iptables etc.? Would that effect other applications on the server? The question is if I can just run it along others on the main server or should I have a dedicated server with PiVPN-wireguard setup?

Also, when I want to remove PiVPN, how do I reset the changes made by the script?

The guide states "The script will also make some changes to your system to allow it to forward internet traffic and allow VPN connections through the Pi's firewall." (https://docs.pivpn.io/) I imagine that any client that connects to the server can get traffic from the internet over the server. But what settings does the server need for this? Would it not be normal anyway that a client coming out of the tunnel uses the server's gateway? 

Thank you!

EDIT: Obviously, I can uninstall with `pipvn -u`!

**Comment by orazioedoardo** [comment, marked as answer]
> What does the script change in terms of iptables etc.?

Adds rules to allow incoming connection on the VPN port (if required), allow forwarding from the virtual interface to physical and reverse (if required), NAT connections from virtual IPs. Not firewall but related, it also changes some sysctls to allow IP forwarding.

> Would that effect other applications on the server?

It should not. It only adds rules to allow, it does not restrict anything.

> I can just run it along others on the main server

Yes.

> Also, when I want to remove PiVPN, how do I reset the changes made by the script?

Just the iptables settings? Remove rules with comment `nameofvpn-something-rule` in `/etc/iptables/rules.v4` and `/etc/iptables/rules.v6`.

> I imagine that any client that connects to the server can get traffic from the internet over the server.

IP forwarding is not allowed on Linux by default for security and avoiding abuse, and even if enabled the server may still block forwarding via the firewall.
